.github/actions/notify_e2e_failure/action.yml

```diff
@@ -59,15 +59,42 @@ runs:
         # Default window: last 7 days
         g='(time:(from:now-7d,to:now))'
 
+        # Query construction
+        # Omit empty fields since OpenSearch will otherwise only display results where the field is empty
+        if [[ -n "${{ inputs.provider }}" ]]
+        then
+          providerQry="(query:(match_phrase:(cloud.provider:${{ inputs.provider }}))),"
+        fi
+        if [[ -n "${{ inputs.refStream }}" ]]
```

Comment:
Maybe this would be easier to maintain as

```sh
query() {
  key=$1
  val=$2
  if [[ -n "$key" ]];  then
    printf "(query:(match_phrase:(%s:'%s')))," "$key" "$val"
  fi
}

q=$(echo "(filters:!(
  $(query metadata.github.ref-stream "${{ inputs.refStream }}")
  # ...
  ))
```